components/dashboard/src/App.tsx

```diff
@@ -4,188 +4,191 @@
-            title: 'Admin',
-            link: '/admin',
-            alternatives: adminMenu.flatMap(e => e.link)
-        });
-    }
-
-    list.push({
-        title: 'Docs',
-        link: 'https://www.gitpod.io/docs/',
-        alternatives: []
-    });
-    list.push({
-        title: 'Community',
-        link: 'https://community.gitpod.io/',
-        alternatives: []
-    });
-
-    return <Menu list={list}/>;
-}
-
-export default App;
+ import React, { Suspense, useContext, useEffect, useState } from 'react';
+ import Menu from './components/Menu';
+ import { BrowserRouter } from "react-router-dom";
+ import { Redirect, Route, Switch } from "react-router";
+ 
+ import { Login } from './Login';
+ import { UserContext } from './user-context';
+ import { getGitpodService } from './service/service';
+ import { shouldSeeWhatsNew, WhatsNew } from './WhatsNew';
+ import settingsMenu from './settings/settings-menu';
+ import { User } from '@gitpod/gitpod-protocol';
+ import { adminMenu } from './admin/admin-menu';
+ 
+ const Workspaces = React.lazy(() => import(/* webpackPrefetch: true */ './workspaces/Workspaces'));
+ const Account = React.lazy(() => import(/* webpackPrefetch: true */ './settings/Account'));
+ const Notifications = React.lazy(() => import(/* webpackPrefetch: true */ './settings/Notifications'));
+ const Plans = React.lazy(() => import(/* webpackPrefetch: true */ './settings/Plans'));
+ const Teams = React.lazy(() => import(/* webpackPrefetch: true */ './settings/Teams'));
+ const EnvironmentVariables = React.lazy(() => import(/* webpackPrefetch: true */ './settings/EnvironmentVariables'));
+ const Integrations = React.lazy(() => import(/* webpackPrefetch: true */ './settings/Integrations'));
+ const Preferences = React.lazy(() => import(/* webpackPrefetch: true */ './settings/Preferences'));
+ const StartWorkspace = React.lazy(() => import(/* webpackPrefetch: true */ './start/StartWorkspace'));
+ const CreateWorkspace = React.lazy(() => import(/* webpackPrefetch: true */ './start/CreateWorkspace'));
+ const InstallGitHubApp = React.lazy(() => import(/* webpackPrefetch: true */ './prebuilds/InstallGitHubApp'));
+ const FromReferrer = React.lazy(() => import(/* webpackPrefetch: true */ './FromReferrer'));
+ const UserSearch = React.lazy(() => import(/* webpackPrefetch: true */ './admin/UserSearch'));
+ const WorkspacesSearch = React.lazy(() => import(/* webpackPrefetch: true */ './admin/WorkspacesSearch'));
+ 
+ function Loading() {
+     return <>
+     </>;
+ }
+ 
+ function App() {
+     const { user, setUser } = useContext(UserContext);
+ 
+     const [loading, setLoading] = useState<boolean>(true);
+     const [isWhatsNewShown, setWhatsNewShown] = useState(false);
+ 
+     useEffect(() => {
+         (async () => {
+             try {
+                 const usr = await getGitpodService().server.getLoggedInUser()
+                 setUser(usr);
+             } catch (error) {
+                 console.log(error);
```

Comment:
<details>
<summary>:warning: <strong>This code seems to be logging information in production - console.log()</strong> (click for details)</summary>

Generally, using `console.log` instead of `alert()` is a good idea for debugging purposes but this can bloat your JavaScript files and also leak out sensitive information of your app as anybody can read application logs by opening dev-tools.   
An important improvement would be to remove this logging statement from the code.  
  
This change would make your app more secure, less prone to information leakage and cleaner.

**[More details](https://confluence.devfactory.com/x/AwIGF)**

## The improvement suggestion in this case
Consider removing `console.log` statements.
</details>

[//]: # (<a id="c76dcba2-2205-4df6-80dd-9599398b5305" data-finding-id="335938051" href="none" codereview-uuid-reserved-tag data-business-id="507be860-41ae-46a0-83e9-7f8e359e3039" data-analyzer="js" />)